noxfile.py

```diff
@@ -0,0 +1,92 @@
+import nox
+import os
+import yaml
+
+@nox.session
+def run_parse(session):
+    session.install('pyyaml')
+    session.cd('src/Build/parse')
+    config_path = session.posargs[0]
+    session.run("python", "parse.py", config_path)
+    session.cd('../../..')
+    
+@nox.session
+def run_generate(session):
+    session.install('pyyaml', 'jinja2')
+    session.cd('src/Build/parse')
+    session.run('python', 'generate.py')
+    session.cd('../../..')
+
+@nox.session
+def build_algorithm(session):
+    """Build the Algorithm docker Image"""
+    if len(session.posargs) > 0:
+        algorithm = session.posargs[0] 
+        print("Algorithm Name in build_algorithm session: ", algorithm)
+    else:
+        algorithm = 'threshold'
+    
+    print("Building Algorithm Nox-File: ", algorithm)
+    image_name = f'{algorithm}'
+    print("Image Name: ", image_name)
+    dockerfile_path = f'src/Algorithms/{algorithm}/Dockerfile'
+    config_file_path = f'src/Algorithms/{algorithm}/config.yaml'
+
+    print("Dockerfile Path: ", dockerfile_path)
+
+    if os.path.exists(dockerfile_path):
+        # If Dockerfile exists, build the image
+        session.run('docker', 'build', '-t', image_name, '-f', dockerfile_path, f'src/Algorithms/{algorithm}')
+    else:
+        # If Dockerfile doesn't exist, look into the config.yaml file
+        print(f'Dockerfile for {algorithm} not found at {dockerfile_path}')
+        
+        if os.path.exists(config_file_path):
+            print(f'Checking config file at {config_file_path}')
+            with open(config_file_path, 'r') as file:
+                config = yaml.safe_load(file)
```

Comment:
See comment on `docker-build.yml`. The `yaml` module is not available here, even if we do `session.install('pyyaml')`. We should refactor this so the logic is run within a `session.run`.